### PR TITLE
fix failures on BSDs by not using fully qualified path

### LINKED ISSUE #434: nodejs::npm::global_config_entry cannot work on a platform that puts npm into a different path
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 6.19.1
- Ruby: 2.6.5
- Distribution: FreeBSD
- Module version: 8.1.0

## How to reproduce (e.g Puppet code you use)
```puppet
  class { 'nodejs':
    nodejs_package_name => 'node12',
    npm_package_name    => 'npm-node12',
    npm_path            => '/usr/local/bin/npm'
  }

  nodejs::npm::global_config_entry { 'unsafe-perm':
    ensure => 'present',
    value  => 'true',
  }
```

## What are you seeing

All of my `nodejs::npm::global_config_entry` are failing, because the code isn't made to run on anything the author didn't anticipate. but it comes down to two things:

```puppet
define nodejs::npm::global_config_entry (
  Enum['present', 'absent'] $ensure = 'present',
  $config_setting                   = $title,
  $cmd_exe_path                     = $nodejs::cmd_exe_path,
  $npm_path                         = $nodejs::params::npm_path,
  $value                            = undef,
) {
  include nodejs
```

`npm_path` defaults to `$nodejs::params::npm_path` rather than `$nodejs::npm_path`, which is wrong.
`$nodejs::params::npm_path` for FreeBSD is also set to `/usr/bin/npm` rather than `/usr/local/bin/npm`.

next, we have a hard-coded path of `/usr/bin/test`, which on FreeBSD is located in `/bin/test`.

Both of these seem… unnecessary. If the `PATH` is correct, we shouldn't have to hard-code path, unless they are *extremely* esoteric.

### PR DESCRIPTION
**Pull request by igalic**
#### Pull Request (PR) description 

if we can't rely on our PATH, we're hosed anyway.

#### This Pull Request (PR) fixes the following issues

Fixes #434

### CODE REVIEW

**Comment by igalic** [comment]
n.b.: I want to add tests to this… for all supported OSes… but OpenBSD & FreeBSD aren't supported right now anyway, so…

File: manifests/npm/global_config_entry.pp [hunk 1]
```diff
@@ -17,13 +17,13 @@
       if $config_setting =~ /(_|:_)/ {
         $onlyif_command = $facts['os']['family'] ? {
           'Windows' => "${cmd_exe_path} /C FOR /F %G IN ('${npm_path} config get globalconfig') DO IF EXIST %G (FINDSTR /B /C:\"${$config_setting}\" %G) ELSE (EXIT 1)",
-          default   => "test -f $(${npm_path} config get globalconfig) && /bin/grep -qe \"^${$config_setting}\" $(${npm_path} config get globalconfig)",
+          default   => "test -f $(${npm_path} config get globalconfig) && grep -qe \"^${$config_setting}\" $(${npm_path} config get globalconfig)",
```
Comment (bastelfreak) [inline manifests/npm/global_config_entry.pp:20]: if we switch to relative paths, do we need to set `$PATH` in the exec resource?

Comment (igalic) [reply manifests/npm/global_config_entry.pp:20]: :+1:

Comment (igalic) [reply manifests/npm/global_config_entry.pp:20]: fixed

**Review by bastelfreak — APPROVED** [review]

File: spec/defines/global_config_entry_spec.rb [hunk 2]
```diff
@@ -11,6 +11,15 @@
         facts
       end
 
+      let(:npm_path) do
+        if os =~ /freebsd/i
```
Comment (igalic) [inline spec/defines/global_config_entry_spec.rb:15]:
```suggestion
        if facts[:os]['family'] == 'FreeBSD'
```

File: spec/defines/nodejs_npm_spec.rb [hunk 3]
```diff
@@ -11,6 +11,14 @@
         facts
       end
 
+      let(:npm_path) do
+        if os =~ /freebsd/i
```
Comment (igalic) [inline spec/defines/nodejs_npm_spec.rb:15]:
```suggestion
        if facts[:os]['family'] == 'FreeBSD'
```

**Review by igalic — COMMENTED** [review]
let's just directly access the facts object

**Review by juniorsysadmin — APPROVED** [review]

File: spec/defines/global_config_entry_spec.rb [hunk 4]
```diff
@@ -11,6 +11,15 @@
...
 
+      let(:npm_path) do
+        if facts[:os]['family'] == 'FreeBSD'
+          '/usr/local/bin/npm'
+        else
+          '/usr/bin/npm'
+        end
+      end
+
+
```
Comment (igalic) [inline spec/defines/global_config_entry_spec.rb:22]:
```suggestion
```

**Review by igalic — COMMENTED** [review]
this line must go to pass rubocop

**Review by bastelfreak — APPROVED** [review]

**Comment by igalic** [comment]
this needs a better changelog entry, so we should at least change the title here: we now actually support FreeBSD, and it's metadata.json and as such it now has tests

**Pull request merged**